xlsx/src/export/worksheets.rs

```diff
@@ -51,15 +53,60 @@ fn get_formula_attribute(
     escape_xml(formula).to_string()
 }
 
-pub(crate) fn get_worksheet_xml(
+pub(crate) fn write_worksheet_xml<W: Write>(
     worksheet: &Worksheet,
     parsed_formulas: &[Node],
     dimension: &str,
     is_sheet_selected: bool,
-) -> String {
-    let mut sheet_data_str: Vec<String> = vec![];
+    writer: &mut W,
+) -> Result<(), XlsxError> {
+    let tab_selected = if is_sheet_selected {
+        " tabSelected=\"1\""
+    } else {
+        ""
+    };
+
+    let show_grid_lines = if !worksheet.show_grid_lines {
+        " showGridLines=\"0\""
+    } else {
+        ""
+    };
+
+    let mut active_cell = "A1".to_string();
+    let mut sqref = "A1".to_string();
+
+    let views = &worksheet.views;
+    if let Some(view) = views.get(&0) {
+        let range = view.range;
+        let row = view.row;
+        let column = view.column;
+        let column_name = number_to_column(column).unwrap_or("A".to_string());
+        active_cell = format!("{column_name}{row}");
+
+        let column_start = number_to_column(range[1]).unwrap_or("A".to_string());
+        let column_end = number_to_column(range[3]).unwrap_or("A".to_string());
+        if range[0] == range[2] && range[1] == range[3] {
+            sqref = format!("{column_start}{}", range[0]);
+        } else {
+            sqref = format!("{}{}:{}{}", column_start, range[0], column_end, range[2]);
+        }
+    }
+
+    write!(
```

Comment:
Write to the zip file in chunks, to avoid memory spikes. But we want the chunks big enough to avoid overhead in `write!` so we'll do some batching below.